### PR TITLE
feat: Added hedera-addressbook-service-impl package for DAB P2

### LINKED ISSUE #13466: Add hedera-addressbook-service-impl package for Dynamic Address Book P2
Create hedera-addressbook-service-impl package for Dynamic Address Book P2

Added Handler classes and their methods which will need to be implemented.
Added all other classes and their testing classes.

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

**Related issue(s)**:

Fixes #13466

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        requireNonNull(feeContext);
+        final var op = feeContext.body();
+
+        // implement
+        return Fees.FREE;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java:77]: Will be nice to `throw new UnSupportedOperationException()` on all the methods until we implement

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        // implement
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+
+        final var op = context.body().nodeDelete();
+        // implement
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java:53]: same here

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        return COST_ANSWER == responseType;
+    }
+
+    @Override
+    public void validate(@NonNull final QueryContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var query = context.query();
+        final var store = context.createStore(ReadableNodeStore.class);
+        final NodeGetInfoQuery op = query.nodeGetInfoOrThrow();
+        // implement
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java:82]: same here and other places

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java]: Updated.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java]: Updated.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java]: Updated.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public Response createEmptyResponse(@NonNull final ResponseHeader header) {
+        requireNonNull(header);
+        final var response = NodeGetInfoResponse.newBuilder().header(header);
+        return Response.newBuilder().nodeGetInfo(response).build();
+    }
+
+    @Override
```
Comment (povolev15) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java:66]: I'm not sure we need to override them. they already implemented in the abstract class of QueryHandler

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
```
Comment (povolev15) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java:72]: Same here

**Review by povolev15 — DISMISSED** [review]

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java (hunk 4 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java:66]: Removed.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java (hunk 5 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java:72]: Removed.

**Review by povolev15 — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @iwsimon

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**